src/app/domain/sharing/services/SharingNetwork.ts

```diff
@@ -0,0 +1,48 @@
+import CozyClient, { generateWebLink, Q } from 'cozy-client'
+
+import { sharingLogger } from '..'
+import { SharingCozyApp } from '../models/SharingCozyApp'
+
+export const fetchSharingCozyApps = {
+  definition: Q('io.cozy.apps').where({
+    'accept_documents_from_flagship.route_to_upload': { $exists: true },
+    accept_from_flagship: true
+  }),
+  options: {
+    as: 'io.cozy.apps/fetchSharingCozyApps'
+  }
+}
+
+export const getRouteToUpload = (
+  cozyApps: SharingCozyApp[],
+  client: CozyClient,
+  session: { subDomainType: string }
+): { href: string; slug: string } | undefined => {
+  try {
+    if (!Array.isArray(cozyApps) || cozyApps.length === 0) return undefined
+
+    const cozyApp = cozyApps.find(cozyApp => cozyApp.slug === 'drive')
+    if (!cozyApp) return undefined
+
+    const hash =
+      cozyApp.accept_documents_from_flagship?.route_to_upload ??
+      cozyApp.attributes.accept_documents_from_flagship?.route_to_upload
+    const slug = cozyApp.slug
+    if (!hash || !slug) return undefined
+
+    const href = generateWebLink({
+      cozyUrl: client.getStackClient().uri,
+      pathname: '',
+      slug,
+      subDomainType: session.subDomainType,
+      hash: hash.replace(/^\/?#?\//, ''),
```

Comment:
Why do we have to make this replace?

Reply:
it's to handle route_to_upload looking like that: `"route_to_upload": "/#/createfromFlagshipUpload=true"`. I did not find a way to handle it directly with the generateWebLink(). I agree it doesn't look great overall. Probably best to remove the leading `/#` in the manifest. I don't think the manifest has to be aware we are using hash router anyway

Reply:
I don't know, actually in the manifest, we can define a few routes. Those routes are used by the server, so if you create a `/test` route, then the url will be : foo.mycozy.cloud/test. 

I don't know if it's a good idea to use `/createfromFlagship` directly in the manifest, since we can think this will be related to the server. 

The manifest is at the intersection between the server and the front end app. 

The manifest is linked to the app, so the manifest knows about the structure of the app and knows if the app use an hash router or not. 

@Ldoppea do you have a better idea?